full-detector-simulations/FCCeeCLD/FCCeeCLD.md

```diff
@@ -23,3 +23,678 @@ source /cvmfs/sw-nightlies.hsf.org/key4hep/setup.sh
+
+<details>
+
+<summary><b>Click to show the code of the python script <code>showPlots.py</code> </b></summary>
+
+```python
+#!/bin/env python
+import sys, ROOT, uproot3 as uproot
+from ROOT import TFile, TTree
+ROOT.gROOT.SetBatch(True)
+if len(sys.argv) < 2:
+  print("Please specify input file"); sys.exit(1)
+inputFile = sys.argv[1]; print("Reading:", inputFile)
+tfile = ROOT.TFile.Open(inputFile)
+myTree = tfile.Get("events")
+myTree.Draw("ECalEndcapCollection.position.z>>zHist(100, 2300, 2510)",
+"ECalEndcapCollection.position.z > 0")
+myTree.Draw("ECalEndcapCollection.energy>>eHist(30, 0, 0.002)")
+zHist = tfile.Get("zHist")
+eHist = tfile.Get("eHist")
+oFile = ROOT.TFile.Open("temp.root", "RECREATE")
+eHist.Write(); zHist.Write(); oFile.Close()
+uFile = uproot.open("temp.root")
+uFile["zHist"].show()
+uFile["eHist"].show()
+```
+
+</details>
+
+<br/>
+
+
+In this section, detector geometry will be changed in every step. Geometry can be visualized using different tools, as it is explained extensively in a following section. 
+
+:::{admonition} Click to show some methods to plot the geometry.
+:class: solution dropdown
+1. The geometry description of the detector in the `xml` file can be converted into ROOT using a python script. The usage of the script is the following: <code>./dd4hep2root -c description.xml -o out.root</code>. The outputfile can be accessed with ROOT, or directly with the ROOT tool `rootbrowse`.
+
+<details>
+
+<summary><b>Click to show the code of the python script <code>dd4hep2root.py</code>. </b></summary>
+
+```python
+#!/usr/bin/env python3
+
+import sys
+import argparse
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Convert detector')
+    parser.add_argument('-c', '--compact', help='Compact file location(s)',
+                        required=True, type=str, nargs='+')
+    parser.add_argument('-o', '--out', help='Converted file path',
+                        default='detector.root', type=str)
+    args = vars(parser.parse_args())
+
+    convert(args['compact'], args['out'])
+
+
+def convert(compact_files, out_path):
+    print('INFO: Converting following compact file(s):')
+    for cfile in compact_files:
+        print('      ' + cfile)
+
+    import ROOT
+
+    ROOT.gSystem.Load('libDDCore')
+    description = ROOT.dd4hep.Detector.getInstance()
+    for cfile in compact_files:
+        description.fromXML(cfile)
+
+    ROOT.gGeoManager.SetVisLevel(9)
+    ROOT.gGeoManager.SetVisOption(0)
+    ROOT.gGeoManager.Export(out_path)
+
+
+if __name__ == '__main__':
+    main()
+
+```
+
+</details>
+
+<br/>
+
+2. Using Qt interface of Geant4, accesible via `runType` argument of `ddsim`
+
+<details>
+
+<summary><b> Click to show how to display the detector geometry using Geant4. </b></summary>
+
+```shell
+source /cvmfs/sft.cern.ch/lcg/views/dev4/latest/x86_64-centos7-gcc11-opt/setup.sh
+ddsim --compactFile description.xml --runType qt
```

Comment:
The example description.xml file should be concrete and copy-pastable for the users.